### PR TITLE
Separate attributes for Method D, I and J

### PR DESCRIPTION
**Pull request by JetteSchumann**
Method options for D, I and J need to be read in and handled separately. Otherwise when Method D, I and J are defined in the same jpsrport ini-file their options are overwritten and the same configuration is used for all of them.

The ArgumentParser and Analysis must be adapted. 

Solves #756

### CODE REVIEW

**Comment by gjaeger** [comment]
@JetteSchumann, I get some errors:
```bash
[ 13%] Building CXX object jpsreport/CMakeFiles/report.dir/Analysis.cpp.o
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/jpsreport/Analysis.cpp:168:37: error: no member named 'GetIndividualFDFlags' in 'ArgumentParser'
        _IndividualFDFlags  = args->GetIndividualFDFlags();
                              ~~~~  ^
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/jpsreport/Analysis.cpp:192:37: error: no member named 'GetIndividualFDFlags' in 'ArgumentParser'
        _IndividualFDFlags  = args->GetIndividualFDFlags();
                              ~~~~  ^
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/jpsreport/Analysis.cpp:197:37: error: no member named 'GetIsCutByCircle' in 'ArgumentParser'
    _cutByCircle            = args->GetIsCutByCircle();
                              ~~~~  ^
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/jpsreport/Analysis.cpp:198:37: error: no member named 'GetIsGetProfile' in 'ArgumentParser'
    _getProfile             = args->GetIsGetProfile();
                              ~~~~  ^
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/jpsreport/Analysis.cpp:199:37: error: no member named 'GetIsOneDimensional' in 'ArgumentParser'
    _isOneDimensional       = args->GetIsOneDimensional();
                              ~~~~  ^
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/jpsreport/Analysis.cpp:202:41: error: no member named 'GetGridSizeX' in 'ArgumentParser'
    _grid_size_X            = int(args->GetGridSizeX());
                                  ~~~~  ^
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/jpsreport/Analysis.cpp:203:41: error: no member named 'GetGridSizeY' in 'ArgumentParser'
    _grid_size_Y            = int(args->GetGridSizeY());
                                  ~~~~  ^
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/jpsreport/Analysis.cpp:207:37: error: no member named 'GetCutRadius' in 'ArgumentParser'
    _cutRadius              = args->GetCutRadius();
                              ~~~~  ^
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/jpsreport/Analysis.cpp:208:37: error: no member named 'GetCircleEdges' in 'ArgumentParser'
    _circleEdges            = args->GetCircleEdges();
                              ~~~~  ^
9 errors generated. 
```

A few warnings:
```bash
[ 21%] Building CXX object libcore/CMakeFiles/core.dir/src/general/ArgumentParser.cpp.o
In file included from /Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/libcore/src/events/TrainEvent.cpp:1:
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/libcore/src/events/TrainEvent.h:29:9: warning: private field '_roomID' is not used [-Wunused-private-field]
    int _roomID;
        ^
/Users/gregorjaeger/Documents/hubs/JuPedSim_fork/jpscore/libcore/src/events/TrainEvent.h:34:9: warning: private field '_subroomID' is not used [-Wunused-private-field]
    int _subroomID;
        ^
2 warnings generated.
```

**Comment by JetteSchumann** [comment]
@gjaeger It is still a draft PR and not working so far.

After talking to @zeroset the data handling should be restructured. A new class holding the information of the configuration options is needed to avoid lots of redundant code.

File: jpsreport/methods/ConfigData_DIJ.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#pragma once
+
+#include <vector>
+
+struct ConfigData_DIJ {
+    //parameters that are provided for measurement areas
+    std::vector<bool> individual_FD_flags;
+    std::vector<int> start_frames;
+    std::vector<int> stop_frames;
+    std::vector<int> areaIDs;
```
Comment (schroedtert) [inline jpsreport/methods/ConfigData_DIJ.h:35]: Is there a mapping from given measurement ID to vector index?

File: jpsreport/methods/ConfigData_DIJ.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    //parameters for creating profiles
+    bool getProfile   = false;
+    float grid_size_X = 0.1;
+    float grid_size_Y = 0.1;
```
Comment (schroedtert) [inline jpsreport/methods/ConfigData_DIJ.h:40]: Default values need to mentioned on website

File: jpsreport/methods/ConfigData_DIJ.h [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    //parameters for cutting voronoi cells
+    bool cutByCircle = false;
+    float cutRadius  = 50;
+    int circleEdges  = 6;
```
Comment (schroedtert) [inline jpsreport/methods/ConfigData_DIJ.h:45]: Default values need to mentioned on website

File: jpsreport/methods/ConfigData_DIJ.h (hunk 1 above)
Comment (zeroset) [reply jpsreport/methods/ConfigData_DIJ.h:35]: Currently, jpsreport reads the measurementIDs for a certain method in this vector and then works on them after each other. There is no real mapping.

**Pull request merged**